src/lib/chat/matrix-client.ts

```diff
@@ -133,7 +133,7 @@ export class MatrixClient implements IChatClient {
   async getMessagesByChannelId(channelId: string, _lastCreatedAt?: number): Promise<MessagesResponse> {
     await this.waitForConnection();
     const { chunk } = await this.matrix.createMessagesRequest(channelId, null, 50, Direction.Backward);
-    const messages = chunk.filter((m) => m.type === EventType.RoomMessage);
+    const messages = chunk.filter((m) => m.type === EventType.RoomMessage && !m?.unsigned?.redacted_because);
```

Comment:
Might be easier to understand if we extract the `m?.unsigned?.redacted_because` to a function `!isDeleted(m)`